src/ucs/datastruct/callbackq.h

```diff
@@ -186,6 +210,43 @@ ucs_status_t ucs_callbackq_remove_safe(ucs_callbackq_t *cbq, ucs_callback_t cb,
 
 
 /**
+ * Add a callback to the slow path list.
+ * This can be used from any context and any thread, including but not limited to:
+ * - A callback can add another callback.
+ * - A thread can add a callback while another thread is dispatching callbacks.
+ *
+ * Important note: It is not allowed to add the same element more than once.
+ *
+ * Complexity: O(n) (n is number of regular callbackq elements)
+ *
+ * @param  [in] cbq      Callback queue to add the callback to.
+ * @param  [in] elem     Slow path list element. The user is expected to
+ *                       initialize the "cb" field.
+ */
+void ucs_callbackq_add_slow_path(ucs_callbackq_t *cbq,
+                                 ucs_callbackq_slow_elem_t* elem);
+
+
+/**
+ * Remove a callback from the slow path list.
+ * This is *not* safe to call while another thread might be dispatching callbacks.
+ * However, it can be used from the dispatch context (e.g a callback may use this
+ * function to remove itself).
+ *
+ * Important note: It is not allowed to pass the element which was not added
```

Comment:
In the function implementation do we have to add #DEBUG code that checks this corner cases ?